pandas/core/common.py

```diff
@@ -2847,11 +2847,11 @@ def _get_handle(path, mode, encoding=None, compression=None):
 
         if compression == 'gzip':
             import gzip
-            f = gzip.GzipFile(path, 'rb')
+            f = gzip.GzipFile(path, mode)
```

Comment:
this defaults to ``rb`` or ``wb`` as appropriate, but *should* this be passed by the caller?